packages/oscal-react-library/src/components/OSCALBackMatter.tsx

```diff
@@ -0,0 +1,178 @@
+import React from "react";
+import { styled } from "@mui/material/styles";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Chip from "@mui/material/Chip";
+import Grid from "@mui/material/Grid";
+import { Typography } from "@mui/material";
+import FormatQuoteIcon from "@mui/icons-material/FormatQuote";
+import OpenInNewIcon from "@mui/icons-material/OpenInNew";
+import StyledTooltip from "./OSCALStyledTooltip";
+import { getAbsoluteUrl, guessExtensionFromHref } from "./oscal-utils/OSCALLinkUtils";
+import { OSCALSection, OSCALSectionHeader } from "../styles/CommonPageStyles";
+import { OSCALMarkupLine } from "./OSCALMarkupProse";
+import OSCALAnchorLinkHeader from "./OSCALAnchorLinkHeader";
+import OSCALEditableTextField from "./OSCALEditableTextField";
+import { Resource, ResourceLink, BackMatter } from "@easydynamics/oscal-types";
+import { ReactElement } from "react";
+
+export const OSCALBackMatterCard = styled(Card)(
+  ({ theme }) => `
+    margin-top: ${theme.spacing(1)};
+    display: flex;
+    flex-direction: column;
+`
+);
+
+function TitleDisplay(props: any): ReactElement {
+  const { children, uuid } = props;
+  return (
+    <OSCALAnchorLinkHeader value={uuid}>
+      <Typography variant="subtitle1">{children}</Typography>
+    </OSCALAnchorLinkHeader>
+  );
+}
+
+function CitationDisplay(props: any): ReactElement {
+  if (!props.resource?.citation?.text) {
+    return (
+      <FormatQuoteIcon
+        color="disabled"
+        fontSize="small"
+        titleAccess={`${props.resource.title}-citation`}
+      />
+    );
+  }
+  return (
+    <StyledTooltip title={<OSCALMarkupLine>{props.resource.citation.text}</OSCALMarkupLine>}>
+      <FormatQuoteIcon
+        color="primary"
+        fontSize="small"
+        titleAccess={`${props.resource.title}-citation`}
+      />
+    </StyledTooltip>
+  );
+}
+
+function getObjectRootKey(object: any): string {
+  if (!object) {
+    return "";
+  }
+
+  const keys = Object.keys(object);
+  if (keys.length !== 1) {
+    throw new Error(
+      `The given object does not have exactly one root key ${JSON.stringify(object)}`
+    );
+  }
+  return keys[0];
+}
+
+interface OSCALBackMatterProps {
+  backMatter: BackMatter | undefined;
+  parentUrl: string;
+  isEditable?: boolean;
+  onFieldSave?: (...args: any[]) => void;
+  partialRestData?: Record<string, any>;
+}
+
+export default function OSCALBackMatter(props: OSCALBackMatterProps): ReactElement {
+  const getMediaType = (rlink: ResourceLink) =>
+    rlink["media-type"] ||
+    guessExtensionFromHref(getAbsoluteUrl(rlink.href, props.parentUrl) ?? "");
+
+  const objectKey = getObjectRootKey(props.partialRestData);
+
+  const backMatterDisplay = (resource: Resource) => (
+    <Grid item xs={3} key={resource.uuid}>
```

Comment:
For composability, rather than being a nested function, this almost feels like it should be its own component in the root of the document...


```tsx
interface EditableFieldProps {
  isEditable?: boolean;
  onFieldSave?: (...args: any[]) => void;
  partialRestData?: Record<string, any>;
}

interface BackMatterResourceProps extends EditableFieldProps {
  resource: Resource;
}

function BackMatterResource(props: BackMatterResourceProps): ReactElement {
  //... the body of this function
}
```

Reply:
I have tried to do this and it is not much cleaner. I think this is becuase you can't really edit a resource title without it being generally aware of all other resources. It may be that I am just missing something, so I will push it and see what everyone thinks.